apps/dsb-message-broker/test/auth.e2e-spec.ts

```diff
@@ -52,17 +56,33 @@ describe('Auth controller (e2e)', () => {
         app.useLogger(['log', 'error']);
 
         await app.init();
+
+        const user = new Wallet(privateKey);
+        identityToken = await generateIdentity(user);
+
+        const userAddress = await user.getAddress();
+        userDID = `did:ethr:${userAddress}`;
     });
 
-    it('should login', async function () {
+    it('should login', function (done) {
         const privateKey = process.env.TEST_LOGIN_PK;
         if (!privateKey) {
             this.skip();
         }
 
-        const user = new Wallet(privateKey);
-        const identityToken = await generateIdentity(user);
+        request(app)
+            .post('/auth/login')
+            .send({ identityToken })
+            .expect(HttpStatus.OK)
+            .then((response) => {
+                const payload = response.body.address + response.body.did + userDID;
+                const digest = utils.id(payload);
+                const address = utils.recoverAddress(digest, response.body.signature);
+
+                assert.equal(address, response.body.address);
 
-        await request(app).post('/auth/login').send({ identityToken }).expect(HttpStatus.OK);
+                done();
```

Comment:
This is ok, but maybe even better would be to await for request

await request(app). ....

Then you can rm done and catch, but anyways just a suggestion.

Reply:
sure, i'll do it for next scenarios